### DISCUSSION TITLE
Mathematical Expressions in RE

### DISCUSSION THREAD

**Discussion by jncengineering** [microsoft/RulesEngine#124] [Q&A]
Is it possible to include mathematical expressions in RE?

Taking the _GiveDiscount10_ rule example, I understand that the "**Expression**" is a **string** evaluated to a **boolean**. Depending on the result, the "**SuccessEvent**" is called or not.
```

{
        "RuleName": "GiveDiscount10",
        "SuccessEvent": "10",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "input1.country == \"india\" AND input1.loyalityFactor <= 2 AND input1.totalPurchasesToDate >= 5000 AND input2.totalOrders > 2 AND input3.noOfVisitsPerMonth > 2"
 }
```
In my particular case, I would like to be able for example, to calculate the **area** of the rectangle from two parameters (**Length** x **Width**) and assign the result to another variable (**Area**) - _Area = Length x Width_.  Something like:

```

{
        "RuleName": "AreaCalculation",
        "SuccessEvent": "area",
        "ErrorMessage": "Length or Width not valid",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "Func<int, int, int> area = (length, width) => length * width"
}
```
Is it possible to assign results of mathematical expressions to other variables for further use using RE?

**Comment by abbasc52** [comment]
If you are looking for assigning a variable to use later, have a look at [ScopedParams](https://microsoft.github.io/RulesEngine/#scopedparams)

If you want custom output, you can use [Post rule execution actions](https://microsoft.github.io/RulesEngine/#post-rule-execution-actions)

Hope this helps 😄
